packages/mars/test/syntax/flatMap.test.ts

```diff
@@ -0,0 +1,19 @@
+import { expectFuture } from '../utils'
+import { flatMap, Future } from '../../src'
+
+describe('FlatMap', () => {
```

Comment:
What about corner cases like:
- no arg
- empty array
- single item

Reply:
no arg is covered by typescript,
empty ✅
single item ✅